lib/ansible/modules/cloud/vmware/vmware_guest_controller.py

```diff
@@ -0,0 +1,553 @@
+                "controller_controllerkey": 100,
+                "controller_devicekey": 1000,
+                "controller_disks_devicekey": [
+                    2000
+                ],
+                "controller_label": "SCSI controller 0",
+                "controller_summary": "LSI Logic SAS",
+                "controller_unitnumber": 3
+            },
+            "1": {
+                "controller_busnumber": 1,
+                "controller_controllerkey": 100,
+                "controller_devicekey": 1001,
+                "controller_disks_devicekey": [],
+                "controller_label": "SCSI controller 1",
+                "controller_summary": "VMware paravirtual SCSI",
+                "controller_unitnumber": 4
+            }
+        },
+        "usb2": {
+            "0": {
+                "controller_busnumber": 0,
+                "controller_controllerkey": 100,
+                "controller_devicekey": 7000,
+                "controller_disks_devicekey": [],
+                "controller_label": "USB Controller",
+                "controller_summary": "Auto connect Disabled",
+                "controller_unitnumber": 22
+            }
+        }
+    }
+"""
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from random import randint
+import time
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task
+
+
+class PyVmomiHelper(PyVmomi):
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+        self.sleep_time = 10
+        self.scsi_device_type = dict(lsilogic=vim.vm.device.VirtualLsiLogicController,
+                                     paravirtual=vim.vm.device.ParaVirtualSCSIController,
+                                     buslogic=vim.vm.device.VirtualBusLogicController,
+                                     lsilogicsas=vim.vm.device.VirtualLsiLogicSASController)
+        self.sata_device_type = vim.vm.device.VirtualAHCIController
+        self.nvme_device_type = vim.vm.device.VirtualNVMEController
+        self.usb_device_type = dict(usb2=vim.vm.device.VirtualUSBController,
+                                    usb3=vim.vm.device.VirtualUSBXHCIController)
+        self.controller_types = dict(self.scsi_device_type, sata=self.sata_device_type, nvme=self.nvme_device_type)
+        self.controller_types.update(self.usb_device_type)
+        self.config_spec = vim.vm.ConfigSpec()
+        self.config_spec.deviceChange = []
+        self.change_detected = False
+        self.disk_ctl_bus_num_list = dict(sata=list(range(0, 4)),
+                                          nvme=list(range(0, 4)),
+                                          scsi=list(range(0, 4)))
+
+    def get_unused_ctl_bus_number(self):
+        """
+        Get gid of occupied bus numbers of each type of disk controller, update the available bus number list
```

Comment:
looks like its gets_rid?

Reply:
Yeah, right, thanks for catching this mistake.